src/ucp/rndv/proto_rndv.c

```diff
@@ -361,14 +361,47 @@ ucp_proto_rndv_ctrl_init(const ucp_proto_rndv_ctrl_init_params_t *params)
     return status;
 }
 
+static size_t
+ucp_proto_rndv_op_thresh(const ucp_proto_init_params_t *init_params,
+                         size_t rma_thresh_config, size_t am_thresh_config)
+{
+    const ucp_proto_select_param_t *select_param = init_params->select_param;
+    uint32_t op_attr_mask = ucp_proto_select_op_attr_from_flags(
+            select_param->op_flags);
+
+    if ((op_attr_mask & UCP_OP_ATTR_FLAG_FAST_CMPL) &&
+        ucs_likely(UCP_MEM_IS_HOST(select_param->mem_type))) {
+        return ucs_min(rma_thresh_config, am_thresh_config);
+    }
+
+    return init_params->worker->context->config.ext.rndv_thresh;
+}
+
+static size_t ucp_proto_rndv_thresh(const ucp_proto_init_params_t *init_params)
+{
+    uint8_t op_id = init_params->select_param->op_id;
+    const ucp_ep_config_t *ep_config =
+            &init_params->worker->ep_config[init_params->ep_cfg_index];
+
+    if ((op_id == UCP_OP_ID_TAG_SEND) || (op_id == UCP_OP_ID_TAG_SEND_SYNC)) {
+        return ucp_proto_rndv_op_thresh(init_params,
+                                        ep_config->tag.rndv.rma_thresh.local,
+                                        ep_config->tag.rndv.am_thresh.local);
```

Comment:
cam we use init_params->worker->context->config.ext.rndv_send_nbr_thresh instead of using ucp_ep_config_t?
since we'd like to remove most of ucp_ep_config_t content when removing protov1